### PR TITLE
fix: default admin defined sender names [experience-5938]

### PR DESCRIPTION
**Pull request by doug-s-nava**
Currently, when setting yourself as a sender and choosing an organization as an admin, the `sender name` of the chosen sender is not set in session storage, and subsequently not supplied to the client header in relevant requests. With this change, when selecting an organization from the admin view as a sender, the sender name will be set as `default`.

Note: as, in non-admin scenarios, where actual senders are logging in with their own credentials, the sender name is sourced directly from the user's okta account. As such, it is not believed that this bug exists in a non-admin context.

Future work: allow for specific selection of different senders within each organization from the admin view, rather than forcing `default`.

## Test Steps:
1. start servers on this branch with `yarn run start:localdev` from `frontend-react` and `./gradlew run` from `prime-router`
2. visit http://localhost:3000/admin/settings logging in as necessary
3. set yourself as a sender and select an organization
4. visit http://localhost:3000/upload
5. open your the network panel of your browser's dev tools
6. upload a csv
7. find the request to `localhost:3000/api/waters`
8. _VERIFY_: the `client` header on your request includes `.default` (see screenshot)

## Screenshot

Screen Shot 2022-06-27 at 11 37 50 AM

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/src/components/Admin/OrgsTable.tsx [hunk 1]
```diff
@@ -30,12 +35,21 @@ export function OrgsTable() {
...
 
     const handleSelectOrgClick = (orgName: string) => {
+        const {
+            state: { active: { senderName, memberType } = {} },
+        } = memberships;
+
+        let payload: Partial<MembershipSettings> = {
+            parsedName: orgName,
+        };
+        if (memberType === "sender") {
```
Comment (kevinhaube) [inline frontend-react/src/components/Admin/OrgsTable.tsx:45]: Use the enum for sender, please

File: frontend-react/src/components/Admin/OrgsTable.tsx [hunk 2]
```diff
@@ -30,12 +35,21 @@ export function OrgsTable() {
     const { memberships } = useSessionContext();
 
     const handleSelectOrgClick = (orgName: string) => {
+        const {
+            state: { active: { senderName, memberType } = {} },
```
Comment (kevinhaube) [inline frontend-react/src/components/Admin/OrgsTable.tsx:39]: That's a pretty confusing destructure; anything against `state.active.memberType`?

**Review by kevinhaube — CHANGES_REQUESTED** [review]
Changes to remove session storage push. Only wanna rely on the context moving forward since it plays nice with React.

File: frontend-react/src/components/Admin/OrgsTable.tsx (hunk 2 above)
Comment (doug-s-nava) [reply frontend-react/src/components/Admin/OrgsTable.tsx:34]: it's more that I like destructuring as a way of declaring in one place at the top of a function what sort of data is needed for the function, and what pieces of data we might need from a complex object. Also like the ability to pluck out multiple pieces of data from a complex object in one go for easy reference later. In this case it's a short function where each variable is only referenced once so it's pretty much a toss up that would come down to stylistic preference, especially if we can use optional chaining.

**Review by kevinhaube — APPROVED** [review]
LGTM!

**Pull request merged**